### PR TITLE
Integration - mesh.py - Unit tests and refactoring for surface/mesh.py

### PR DESCRIPTION
**Pull request by emmanesbit**

### CODE REVIEW

File: tests/test_surface.py [hunk 1]
```diff
@@ -329,6 +324,53 @@ def test_explicit_mesh(example_model_and_crs):
     assert_array_almost_equal(persistent_mesh.full_array_ref(), mesh.full_array_ref())
 
 
+@pytest.mark.parametrize('flavour,infile,filetype', [('explicit', 'Surface_roxartext.txt', 'roxar'),
```
Comment (andy-beer) [inline tests/test_surface.py:327]: We should check that there are no IP issues with regard to including test files in these formats in the public repository.

**Review by andy-beer — APPROVED** [review]
This all looks fine with a quick look.

File: resqpy/surface/mesh.py [hunk 2]
```diff
@@ -133,104 +133,201 @@ def __init__(self,
...
+            self._load_from_xyz_values(xyz_values)
+
+        else:
+            self._load_from_arguments(ni, nj, origin, dxyz_dij, z_values, z_supporting_mesh_uuid, crs_uuid)
 
         assert self.crs_uuid is not None
         if not self.title:
             self.title = 'mesh'
 
+    def _load_from_arguments(self, ni, nj, origin, dxyz_dij, z_values, z_supporting_mesh_uuid, crs_uuid):
```
Comment (cflynn3) [inline resqpy/surface/mesh.py:148]: Please use double underscore for private methods / properties.

**Review by cflynn3 — APPROVED** [review]
Aside from the naming issue mentioned this looks like a great set of changes.

**Pull request merged**